crates/parol_runtime/src/parser/parser_types.rs

```diff
@@ -408,20 +427,19 @@ impl<'t> LLKParser<'t> {
                         self.production_depth -= 1;
                         trace!("Popped production {} -> depth {}", p, self.production_depth);
                         self.parser_stack.stack.pop(); // Pop the End of production marker
-                        self.process_item_stack(&mut tree_builder, p, user_actions)?;
+                        self.process_item_stack(tree_builder, p, user_actions)?;
                     }
                 }
             }
         }
-
+        // Handle additional tokens after the last token relevant for the grammar
+        self.handle_additional_tokens(tree_builder, stream.clone(), user_actions)?;
         if !self.error_entries.is_empty() {
             return Err(ParserError::SyntaxErrors {
                 entries: self.error_entries.drain(..).collect(),
             }
             .into());
         }
-        // Handle additional tokens after the last token relevant for the grammar
-        self.handle_additional_tokens(&mut tree_builder, stream.clone(), user_actions)?;
```

Comment:
This is a breaking change. EOF whitespaces and comments are now included in the output tree even if syntax errors happen.